posts/2022-06-09-time-based-log-rollover-22007-beta.adoc

```diff
@@ -0,0 +1,146 @@
+
+* `rolloverInterval` is the desired time interval between log rollovers. Set the attribute with a positive integer followed by a unit of time, which can be days (d), hours (h), or minutes (m). For example, specify 5 hours as `5h``. You can include multiple values in a single entry. For example, `1d5h`` is equivalent to 1 day and 5 hours.
+
+* `rolloverStartTime` is the time of day when logs first roll over and the first rollover interval duration begins. Valid values for this attribute follow a 24-hour ISO-8601 date and time format of HH:MM, where 00:00 represents midnight. 
+
+The two configuration attributes are optional. If neither of the attributes are set, then time based log rollover is not enabled. If only one of the two attributes is set, then time based log rollover is enabled, and the other attribute is set to its default. The default value for `rolloverInterval` is `1d`, and the default value for `rolloverStartTime` is `00:00`.
+
+Example `server.xml` file configuration for log rollover in the `messages.log` and `trace.log` files set by specifying the new attributes in the `logging` element:
+
+[source, xml]
+----
+<server>
+   ...
+    <logging rolloverStartTime="00:00" rolloverInterval="1d"/>
+    ...
+</server>
+----
+
+Example `server.xml` file configuration for log rollover in the `http_access.log` file set by specifying the new attributes in the `httpAccessLogging` element:
+
+[source, xml]
+----
+<server>
+   ...
+    <httpAccessLogging rolloverStartTime="00:00" rolloverInterval="1d" logFormat='%h %u %{t}W "%r" %s %b' />
+    ...
+</server>
+----
+
+Example `server.xml` configuration for access logging, using `<accessLogging/>` under `<httpEndpoint/>`, which handles `http_access.log` rollover:
+
+[source, xml]
+----
+<server>
+   ...
+   <httpEndpoint id="defaultHttpEndpoint">
+      <accessLogging rolloverStartTime="00:00" rolloverInterval="1d" logFormat='%h %i %u %t "%r" %s %b' />
+   </httpEndpoint>
+    ...
+</server>
+----
+
+Example result of logs, rolled over every minute, with a configuration of `rolloverInterval="1m"` and `rolloverStartTime="00:00"`:
+
+image::/img/blog/log_rollover.png[Example logs,width=70%,align="left"]
+
+For more information about this feature, refer to the link:{url-prefix}/docs/latest/reference/config/httpAccessLogging.html[HTTP Access Logging], link:{url-prefix}/docs/latest/reference/config/logging.html[Liberty Logging], and link:{url-prefix}/docs/latest/access-logging.html[Liberty Access Logging] documentation.
+
+[#logrecordcontext]
+== Add application name to LogRecordContext Extension and as JSON logging field
+
+Previously, Liberty logs did not contain the application name in its `LogRecordContext` extension. If you were using a log analysis tool, such as the ELK stack, you couldn't filter out application logs, since the `JSON` fields did not have a field for the application name. 
```

Comment:
```suggestion
Previously, Liberty logs did not contain the application name in the `LogRecordContext` extension. If you were using a log analysis tool, such as the ELK stack, you couldn't filter out application logs, since the `JSON` fields did not have a field for the application name. 
```